File: Python&Ruby/OOP3.rb
class C
  def initialize(n1)
  # 생성자
    @value = n1
  end
  def show()
    puts(@value)
  end
  def getValue()
    return @value
  end
  def setValue(n1)
  # value값 지정을 위해 ()에 n1 입력
    @value = n1
  end
end

c1 = C.new(10)
puts(c1.getValue()) # 결과 : 10
# return @value으로 인해 C.new(10)으로 정한 값인 10 출력

c1.setValue(100)
puts(c1.getValue()) # 결과 : 100
# c1.setValue(100)을 통해 100을 value값으로 지정 후 출력

c1.value = 1000 # 결과 : Error
puts(c1.value) # 결과 : Error
# 외부에서 인스턴스 내부의 메소드에 접근 불가

File: Python&Ruby/OOP3.py
#!python
print("Content-type: text/html; charset=utf-8\n")
# ↑ Python을 연동시켜서 화면에 도출될 수 있게 하는 코드 ↑ (1~2)

class C:
    def __init__(self, n1):
    # 생성자
        self.value = n1
    def show(self):
        print(self.value)
    def getValue(self):
        return self.value
    def setValue(self, n1):
    # value값 지정을 위해 ()에 n1 입력
        self.value = n1

c1 = C(10)
# c1은 인스턴스를 의미, (10)은 위의 self.value 값
print(c1.getValue()) # 결과 : 10
# return self.value으로 인해 C(10)으로 정한 값인 10 출력

c1.setValue(100)
print(c1.getValue()) # 결과 : 100
# c1.setValue(100)을 통해 100을 value값으로 지정 후 출력
